sources/packages/backend/libs/integrations/src/services/disbursement-schedule/e-cert-processing-steps/validate-cslp-lifetime-maximum-check-step.ts

```diff
@@ -0,0 +1,56 @@
+import { Injectable } from "@nestjs/common";
+import { ECertProcessStep } from "./e-cert-steps-models";
+import { ProcessSummary } from "@sims/utilities/logger";
+import { EntityManager } from "typeorm";
+import { EligibleECertDisbursement } from "../disbursement-schedule.models";
+import { ECertGenerationService } from "../e-cert-generation.service";
+import { CANADA_STUDENT_LOAN_PART_TIME_AWARD_CODE } from "@sims/services/constants";
+import { StudentLoanBalanceSharedService } from "@sims/services/student-loan-balance/student-loan-balance-shared.service";
+import { StudentLoanBalance } from "@sims/sims-db";
+
+/**
+ * Fetch the latest  balance for the student and validate if CSLP has reached lifetime maximum.
+ */
+@Injectable()
+export class ValidateCSLPLifetimeMaximumCheckStep implements ECertProcessStep {
```

Comment:
@dheepak-aot @andrepestana-aot @sh16011993 @lewischen-aot @katrina-aot-git 
Just to put everybody on the same page, the initial idea was to build the validation inside the step that is already in place and was mentioned in the technical details.
The PR is proposing a new step exclusive for part-time that will be executed right after the other validations.
It seems to be ok to have it separated from the code organization perspective. Does anyone have any concerns about it?

Reply:
From my understanding of the Ecert processing framework, every step in the processing has it's own logical responsibility. For instance, in PT
The steps are
1. Validate disbursements to check if they are eligible for Ecert processing
2. Apply overawards 
3. Calculate effective values
4. Calculate effective tuition remittance
5. Create BC Totals
6. Save calculations

Hence the step, `Validate disbursement` is one responsibility and IMO we don't have to create more steps for the same purpose(validation). For segregation of each validation that happens inside the `validate disbursement` private methods could be helpful.

I would recommend adding the new validation to the existing step.